kiwi/xml_state.py

```diff
@@ -2212,6 +2213,39 @@ def get_fs_create_option_list(self) -> List:
 
         return option_list
 
+    def get_luks_credentials(self) -> Optional[str]:
+        """
+        Return key or passphrase credentials to open the luks pool
+
+        :return: data
+
+        :rtype: str
+        """
+        data = self.build_type.get_luks()
+        if data and data.startswith('file://'):
```

Comment:
yeah we even have an Uri class. I did not use it because this imply we allow keyfiles from other locations like http, ftp, etc etc. As this is the path to a private key file it will most likely never be exposed on anything else than the local system. That was the reason why I chose only file://

I'm open to change that if people thinks it makes sense ?

Reply:
I think we should handle it properly. Maybe extend the Uri class to have a flag to restrict to local only as desired?

Reply:
`/me` wish to have such hadouken powers of Conan-Kudo over @schaefi sometimes... :wink:

Reply:
This are just the simple rules of democracy :smile:

Reply:
changes done, this is now hopefully more acceptable :) Thanks